src/ripple/app/misc/impl/AccountTxPaging.cpp

```diff
@@ -0,0 +1,246 @@
+        R"(SELECT AccountTransactions.LedgerSeq,AccountTransactions.TxnSeq,
+          Status,RawTxn,TxnMeta
+          FROM AccountTransactions INNER JOIN Transactions
+          ON Transactions.TransID = AccountTransactions.TransID
+          AND AccountTransactions.Account = '%s' WHERE
+          )");
+
+    std::string sql;
+
+    if (forward && (findLedger == 0))
+    {
+        sql = boost::str (boost::format(
+            prefix +
+            (R"(AccountTransactions.LedgerSeq BETWEEN '%u' AND '%u'
+             ORDER BY AccountTransactions.LedgerSeq ASC,
+             AccountTransactions.TxnSeq ASC
+             LIMIT %u;)"))
+            % account.humanAccountID()
+            % minLedger
+            % maxLedger
+            % queryLimit);
+    }
+    else if (forward && (findLedger != 0))
+    {
+        sql = boost::str (boost::format(
+            prefix +
+            (R"(
+            AccountTransactions.LedgerSeq BETWEEN '%u' AND '%u' OR
+            ( AccountTransactions.LedgerSeq = '%u' AND
+              AccountTransactions.TxnSeq >= '%u' )
+            ORDER BY AccountTransactions.LedgerSeq ASC,
+            AccountTransactions.TxnSeq ASC
+            LIMIT %u;
+            )"))
+        % account.humanAccountID()
+        % (findLedger + 1)
+        % maxLedger
+        % findLedger
+        % findSeq
+        % queryLimit);
+    }
+    else if (!forward && (findLedger == 0))
+    {
+        sql = boost::str (boost::format(
+            prefix +
+            (R"(AccountTransactions.LedgerSeq BETWEEN '%u' AND '%u'
+             ORDER BY AccountTransactions.LedgerSeq DESC,
+             AccountTransactions.TxnSeq DESC
+             LIMIT %u;)"))
+            % account.humanAccountID()
+            % minLedger
+            % maxLedger
+            % queryLimit);
+    }
+    else if (!forward && (findLedger != 0))
+    {
+        sql = boost::str (boost::format(
+            prefix +
+            (R"(AccountTransactions.LedgerSeq BETWEEN '%u' AND '%u' OR
+             (AccountTransactions.LedgerSeq = '%u' AND
+              AccountTransactions.TxnSeq <= '%u')
+             ORDER BY AccountTransactions.LedgerSeq DESC,
+             AccountTransactions.TxnSeq DESC
+             LIMIT %u;)"))
+            % account.humanAccountID()
+            % minLedger
+            % (findLedger - 1)
+            % findLedger
+            % findSeq
+            % queryLimit);
+    }
+
+    auto db (connection.checkoutDb());
+
+    Blob rawData;
+    Blob rawMeta;
+
+    boost::optional<std::uint64_t> ledgerSeq64;
+    boost::optional<std::uint32_t> txnSeq;
+    boost::optional<std::string> status;
+    soci::blob txnData (*db);
+    soci::blob txnMeta (*db);
+    soci::indicator dataPresent, metaPresent;
+
+    soci::statement st = (db->prepare << sql,
+        soci::into (ledgerSeq64),
+        soci::into (txnSeq),
+        soci::into (status),
+        soci::into (txnData, dataPresent),
+        soci::into (txnMeta, metaPresent));
+
```

Comment:
It would be nice if we could use `soci::use` instead of `boost::format`. However, I don't know a nice way to handle the different number of `use` parameters used in this sql statement. The way soci executes statements on the destruction of a temporary makes solving this problem more complicated than just using `boost::format`.